packages/compass-query-bar/src/components/hooks.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useCallback, useContext, useMemo } from 'react';
+import type QueryBar from './query-bar';
+import { useSelector, useStore } from '../stores/context';
+import type { ChangeFilterEvent } from '../modules/change-filter';
+import { applyFilterChange } from '../stores/query-bar-reducer';
+import { mapFormFieldsToQuery } from '../utils/query';
+
+const QueryBarComponentContext = React.createContext<typeof QueryBar>(
+  (() => null) as unknown as typeof QueryBar
+);
```

Comment:
incredibly optional: would it make sense to provide a dummy component just so it shows up in e.g. React DevTools? I could see myself getting quite confused about why no query bar component is rendering where I expect it to

```suggestion
function NoOpQueryBarComponent() {
  return <></>;
}

const QueryBarComponentContext = React.createContext<
  React.FC<React.ComponentProps<typeof QueryBar>>
>(NoOpQueryBarComponent);
```

Reply:
Yes, great suggestion, didn't thought about that aspect, thank you!